pulpcore/tasking/tasks.py

```diff
@@ -55,11 +62,15 @@ def _acquire_worker(resources):
     workers_qs = Worker.objects.online_workers().exclude(
         name=constants.TASKING_CONSTANTS.RESOURCE_MANAGER_WORKER_NAME
     )
-    workers_qs_with_counts = workers_qs.annotate(models.Count("reservations"))
+    workers_qs_no_res = (
+        workers_qs.annotate(models.Count("reservations"))
+        .filter(reservations__count=0)
+        .order_by("?")
+    )
     try:
         # A randomly-selected Worker instance that has zero ReservedResource entries
         # associated with it.
-        return workers_qs_with_counts.filter(reservations__count=0).order_by("?")[0]
+        return Worker.objects.select_for_update().filter(pk__in=workers_qs_no_res).first()
```

Comment:
I guess the selected worker isn't necessarily random anymore.

Reply:
Let's fix that. What if we did this:

```python
workers_qs_no_res = (
        workers_qs.annotate(models.Count("reservations"))
        .filter(reservations__count=0)
        .order_by("?")[0:1]
    )
```

Reply:
Actually I think this is better:

```suggestion
        return Worker.objects.select_for_update().filter(pk__in=workers_qs_no_res[0:1]).first()
```